utils/backoff.go

```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/cenkalti/backoff/v4"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// WithMaxRetriesLog runs the operation until it succeeds or max retries has been reached.
+// It uses exponential back off.
+// It optionally logs information if logger is set.
+func WithMaxRetriesLog(
+	operation backoff.Operation,
+	max uint64,
+	logger logging.Logger,
+	msg string,
+	fields ...zapcore.Field,
```

Comment:
I think logging `msg` and `fields` on each attempt is not worth the added complexity of having to pass them in as arguments. Rather, `WithMaxRetries` should emit a generic warning log on each attempt failure, and we can leave it to the caller to construct a cohesive error log in the failure case.